### DISCUSSION TITLE
@Retryable at both class and method level

### DISCUSSION THREAD

**Discussion by mittalarpan** [spring-projects/spring-retry#325] [Q&A] [answered]
If we have `@Retryable` at both class and method level, which one takes precedence for that method ? Based on my testing I believe method level takes precedence but wanted to confirm this with community. Thanks. 

Example - 
```
@FeignClient(name = "name", url = "${url}")
@Retryable(maxAttempts = 4, backoff = @Backoff(delay = 500L, maxDelay = 20_000L, multiplier = 4, random = true),
           include = RetryableException.class)
public interface ClientService {
   
   @Retryable(maxAttempts = 2, backoff = @Backoff(delay = 100L, maxDelay = 10_000L, multiplier = 2, random = true),
           include = RetryableException.class)
   Response getFoo(@RequestParam("param") String param);
}
```

For method `getFoo` I believe the config specified at method level will be used rather than the class level ?

**Comment by garyrussell** [comment, marked as answer]
That is correct; logic is here:

https://github.com/spring-projects/spring-retry/blob/61960f36e720afe8845f9a848a46b64b7216df41/src/main/java/org/springframework/retry/annotation/AnnotationAwareRetryOperationsInterceptor.java#L175-L178

**Reply by mittalarpan** [reply]
Thanks for confirming.
